pkg/templates/promrules.go

```diff
@@ -130,3 +145,61 @@ func (a *PrometheusRuleTemplateManager) CreateFromIngress(ingress *extensionsv1b
 
 	return collectPrometheusRules(prometheusRules), nil
 }
+
+// CreateFromDeployment
+// - Creates all the promRules for a given Deployment
+func (a *PrometheusRuleTemplateManager) CreateFromDeployment(deployment *apps.Deployment, depNamespacePrometheus string) ([]*monitoringv1.PrometheusRule, error) {
+	deploymentIdentifier := fmt.Sprintf("%s.%s", deployment.Namespace, deployment.Name)
+
+	params := &templateParameterDeployment{
+		Deployment:   deployment,
+		Identifier:   deploymentIdentifier,
+		Namespace:    deployment.Namespace,
+		Name:         deployment.Name,
+		Selector:     deployment.GetLabels()["app"],
```

Comment:
This seems suspect, why are we not using the entire `deployment.Spec.Selector` and turning that into the selector for metric in the template?